nucliadb_node/src/writer/mod.rs

```diff
@@ -241,4 +247,54 @@ impl NodeWriterService {
             .collect();
         ShardIds { ids }
     }
+
+    #[tracing::instrument(skip_all)]
+    pub async fn move_shard(
```

Comment:
I dont think this logic should be in `NodeWriterService`, it is `async` and is tightly related with our grpc interface.  Maybe it would be better to move it to `NodeWriterGRPCDriver::move_shard`.